#==================================================================================================
# Upstream bug fix, for a few fundamental issues with this source file. Removed in gtk3 and
# later, so only applicable to gtk2.
#
# Date: 2024-04-25
# See: https://trac.macports.org/ticket/69471
#==================================================================================================
--- tests/testmenubars.c.orig	2024-04-24 14:14:19.000000000 -0400
+++ tests/testmenubars.c	2024-04-24 14:17:42.000000000 -0400
@@ -21,7 +21,7 @@
 #include <gtk/gtk.h>
 
 static GtkWidget *
-create_menu (depth)
+create_menu (int depth)
 {
     GtkWidget *menu;
     GtkWidget *menuitem;
@@ -70,19 +70,19 @@
   menuitem = gtk_image_menu_item_new_from_stock (GTK_STOCK_HOME, NULL);
   gtk_menu_shell_append (GTK_MENU_SHELL (menubar), menuitem);
   gtk_label_set_angle (GTK_LABEL (GTK_BIN (menuitem)->child), angle);
-  menu = create_menu (2, TRUE);
+  menu = create_menu (2);
   gtk_menu_item_set_submenu (GTK_MENU_ITEM (menuitem), menu);
 
   menuitem = gtk_menu_item_new_with_label ("foo");
   gtk_menu_shell_append (GTK_MENU_SHELL (menubar), menuitem);
   gtk_label_set_angle (GTK_LABEL (GTK_BIN (menuitem)->child), angle);
-  menu = create_menu (2, TRUE);
+  menu = create_menu (2);
   gtk_menu_item_set_submenu (GTK_MENU_ITEM (menuitem), menu);
 
   menuitem = gtk_menu_item_new_with_label ("bar");
   gtk_menu_shell_append (GTK_MENU_SHELL (menubar), menuitem);
   gtk_label_set_angle (GTK_LABEL (GTK_BIN (menuitem)->child), angle);
-  menu = create_menu (2, TRUE);
+  menu = create_menu (2);
   gtk_menu_item_set_submenu (GTK_MENU_ITEM (menuitem), menu);
 
   return menubar;
